### ISSUE TITLE
Add "reply, delete and edit message"

### ISSUE THREAD

**Issue by wuniversales** [qTox/qTox#5902]
OS: Linux Manjaro 18.1.1
qTox version: v1.16.3.
Commit hash:  8eed684c37bd711de2ca4bd863f5f0509edcdc2d
toxcore:  0.2.10
Qt:  5.12.0
------
Hello,
It occurs to me that it would be a good idea to be able to delete, modify, and reply to messages previously sent in the chat.
Something like telegram. :)
Regards

**Comment by anthonybilinski** [comment]
"delete" is kind of impossible to implement in a reliable way, since even if you ask your peer to delete their history, they could just ignore the request, and if someone asks you do delete your history, why should you listen :P

The general idea of those functionality is definitely desirable, and with e.g. message editing we could accept the edits but still preserve the old history in a nice way where the user can view edit history, so that it's a win/win interaction.

Unfortunately the Tox protocol doesn't support any kind of edit/modify/quote messages that can be sent, or long term IDs of messages that could be used to reference past messages, so it's not possible to add any of that functionality in qTox at this time. 

If Tox adds the packets types to do that in the future, or more likely if [ToxExt](https://github.com/toxext/toxext) continue work to extend the Tox protoco with that functionalityl, we'll add this functionality to qTox at that time.

**Comment by wuniversales** [comment]
Yes, I understand, but even if the partner ignores this request to eliminate something, I think that aesthetically there would be a cleaner chat for both users.

For example, if I forget a question mark, or an accent, it is always better to modify the previous message to both users, even if the partner ignores the request, and keeps a copy of the old unedited message ...

This is a cleaner chat, I'm talking at the aesthetic level.
At the security level, if some improvement in tox can be implemented in future versions, it would be great.

Regards
